### ISSUE TITLE
Internal server error while retrieving secrets

### ISSUE THREAD

**Issue by Shawe82** [PrefectHQ/prefect#6056]
## Description
In our workflow, we use slack notifications to alert us on failing tasks and flows. For that, we use the `slack_notifier` function shipped with prefect. This notifier tries to get a slackhook URL from a secret using the `prefect.client.Secret`. Now, occasionally, this raises an error that lets the entire flow fail. This is especially annoying in case it happens for a task that actually didn't fail, which means the notifier which would not have been shot any notification fails the entire flow. The stack trace tells the following. Note that most of the time, this works a it is, mean this secret actually exists. 

```
Exception raised while calling state handlers: ClientError([{'path': ['secret_value'], 'message': 'An unknown error occurred.', 'extensions': {'code': 'INTERNAL_SERVER_ERROR'}}])
Traceback (most recent call last):
  File "/opt/app/.venv/lib/python3.10/site-packages/prefect/client/secrets.py", line 140, in get
    value = secrets[self.name]
KeyError: 'slack_webhook_data_prd'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/opt/app/.venv/lib/python3.10/site-packages/prefect/engine/cloud/flow_runner.py", line 119, in call_runner_target_handlers
    new_state = super().call_runner_target_handlers(
  File "/opt/app/.venv/lib/python3.10/site-packages/prefect/engine/flow_runner.py", line 116, in call_runner_target_handlers
    new_state = handler(self.flow, old_state, new_state) or new_state
  File "/opt/app/.venv/lib/python3.10/site-packages/toolz/functoolz.py", line 306, in __call__
    return self._partial(*args, **kwargs)
  File "/opt/app/.venv/lib/python3.10/site-packages/prefect/utilities/notifications/notifications.py", line 299, in slack_notifier
    str, prefect.client.Secret(webhook_secret or "SLACK_WEBHOOK_URL").get()
  File "/opt/app/.venv/lib/python3.10/site-packages/prefect/client/secrets.py", line 164, in get
    raise exc
  File "/opt/app/.venv/lib/python3.10/site-packages/prefect/client/secrets.py", line 148, in get
    result = self.client.graphql(
  File "/opt/app/.venv/lib/python3.10/site-packages/prefect/client/client.py", line 464, in graphql
    raise ClientError(result["errors"])
prefect.exceptions.ClientError: [{'path': ['secret_value'], 'message': 'An unknown error occurred.', 'extensions': {'code': 'INTERNAL_SERVER_ERROR'}}]
```

## Expected Behavior
No notification, or retries but not letting the entire flow crash.

## Environment
Prefect cloud, python 3.10, prefect 1.2.4

**Comment by jbnitorum** [comment]
We are seeing a similar issue where we see the following message as our Last State Message on the UI.  We use GitLab storage for our flows and this error is occurring before any tasks inside the flow run.  I assume it must have hit the error when using our secret to access GitLab.  This doesn't happen every run, far from it, in the course of a few thousand flow runs we have seen this handful of times.

` [23 July 2022 6:00am]: Failed to load and execute flow run: ClientError([{'path': ['secret_value'], 'message': 'An unknown error occurred.', 'extensions': {'code': 'INTERNAL_SERVER_ERROR'}}]) `

**Comment by dkomisar** [comment]
We're seeing this error too https://cloud.prefect.io/anaconda/flow-run/553168a0-a2cc-4747-8544-e3c63b3db36c is an example.

**Comment by madkinsz** [comment]
I believe we have resolved issues with the secret service that was causing this. Let me know if this is still occurring.

**Comment by jbnitorum** [comment]
We had another instance of this issue at 8:01pm.


https://cloud.prefect.io/nitorumcapital/flow-run/3cf7985b-d2b9-46ad-9937-f3062205187b?logs


James Brink
Head Trader & Head of Data Research
Office: (212) 356-6315
Cell: (646) 477-9120
***@***.******@***.***>

Nitorum Capital, L.P.
450 Park Ave, 7th Floor
New York, New York 10022

On Sep 26, 2022, at 10:33 AM, Michael Adkins ***@***.******@***.***>> wrote:



I believe we have resolved issues with the secret service that was causing this. Let me know if this is still occurring.

—
Reply to this email directly, view it on GitHub<https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_PrefectHQ_prefect_issues_6056-23issuecomment-2D1258133429&d=DwMCaQ&c=euGZstcaTDllvimEN8b7jXrwqOf-v5A_CdpgnVfiiMM&r=gJzKLxmZPmrTTxFdqQ23VzWC6QhGEqFftN04JwGaJ9I&m=TrGW1MGTPiM7UJ3h5QzrcmTALI8U37OmrtZDG6pNrGI&s=nZeWLm2RWUMAODwlOyBVeaOPOeqejgfVMtX0yAIag1g&e=>, or unsubscribe<https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_notifications_unsubscribe-2Dauth_ACLFKFMKMFJB76ZHVUWNFJDWAGX2HANCNFSM54RSOAKA&d=DwMCaQ&c=euGZstcaTDllvimEN8b7jXrwqOf-v5A_CdpgnVfiiMM&r=gJzKLxmZPmrTTxFdqQ23VzWC6QhGEqFftN04JwGaJ9I&m=TrGW1MGTPiM7UJ3h5QzrcmTALI8U37OmrtZDG6pNrGI&s=1QBZeKPbyw9g8yfSp9_tjf3A0F_6dPg_ePSDdkvco2c&e=>.
You are receiving this because you commented.Message ID: ***@***.***>



The information contained in this email message is sent on a confidential basis to the recipient(s) named above. If you are not the intended recipient or an agent responsible for delivering it to the intended recipient, you have received this email message in error and any review, dissemination, distribution, or copying of this email message is strictly prohibited. If you have received this email message in error, please notify us immediately by email and delete the original message. This email message (including any attachment) is not an offer to sell you any securities or investment advisory services. Any offering of interests in a fund that we manage is made only pursuant to a private placement memorandum and such interests may be purchased only by a qualified person pursuant to subscription documents that must be read and completed in their entirety. Any past performance provided in this email or any attachment is not necessarily indicative of future results. Unless otherwise noted, all performance figures provided are net of expenses, including management fees, brokerage commissions, and performance allocations, and reflect the reinvestment of all gains and other income.

This email message is intended exclusively for the use of the recipient(s) named above, and it is not to be reproduced or redistributed to any other person without the prior written consent of Nitorum Capital, L.P.

**Comment by jbnitorum** [comment]
Another instance this morning at 8:01am…  it seems to be occurring at 8am and 8pm, not sure if that’s helpful at all.

https://cloud.prefect.io/nitorumcapital/flow-run/08384970-4fab-4761-90a7-f8750ca2b7fe?logs

**Comment by madkinsz** [comment]
Thanks for the reports. We're aware of a new issue and are attempting to identify the cause.

**Comment by jbnitorum** [comment]
Hello James,

Thanks for sending this over.
We'll share this information with the team.
We're updating our status page as the investigation progresses.

Warm Regards,
Prefect Support


--------------- Original Message ---------------
From: James Brink ***@***.***
Sent: 9/27/2022 8:17 AM
To: ***@***.***
Cc: ***@***.***; ***@***.***; ***@***.***
Subject: RE: [PrefectHQ/prefect] Internal server error while retrieving secret in slack notifier (Issue #6056)

Another instance this morning at 8:01am  it seems to be occurring at 8am and 8pm, not sure if thats helpful at all.


https://cloud.prefect.io/nitorumcapital/flow-run/08384970-4fab-4761-90a7-f8750ca2b7fe?logs

**Comment by madkinsz** [comment]
We are continuing to work on stabilize the service backing the secrets API. In the meantime, we will be deploying changes to Cloud that facilitate client-side retries on this kind of failure.
